tests/integration-tests/tests/createami/test_createami.py

```diff
@@ -420,6 +430,29 @@ def test_build_image_custom_components(
     _test_build_image_success(image)
 
 
+def _test_build_imds_settings(image, status, region):
+    logging.info(f"Checking that the ImageBuilder instances have IMDSv2 {status}")
+
+    instance_names = [
+        f"Build instance for ParallelClusterImage-{image.image_id}",
+        f"Test instance for ParallelClusterImage-{image.image_id}",
+    ]
+
+    describe_response = boto3.client("ec2", region_name=region).describe_instances(
+        Filters=[{"Name": "tag:Name", "Values": instance_names}]
+    )
+
+    for reservations in describe_response.get("Reservations"):
+        for instance in reservations.get("Instances"):
+            instance_id = instance.get("InstanceId")
+            instance_name = instance.get("Tags").get("Name")
+            imds_v2_status = instance.get("MetadataOptions").get("HttpTokens")
+
+            logging.info(f"Instance {instance_id} ({instance_name}) has IMDSv2 {imds_v2_status}")
+
+            assert_that(imds_v2_status).is_equal_to(status)
```

Comment:
This assertion logic is duplicated. what about defining it in a single common place and reuse it?

Reply:
Yes, in this case I had a slightly different logging message because the ImageBuilder instances have the "Test"/"Build" embedded in the name, but it can be refactored to extract this common logic. Is this a blocking issue, or can it be done in a follow up PR?

Reply:
I suggest to remove the code duplication in this PR to not forget about it.
In any case this is not a blocking comment.

Reply:
Agreed, sending change in this PR.